junit5/src/main/java/com/linecorp/armeria/testing/junit5/client/HttpDataAssert.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.client;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.charset.Charset;
+
+import com.linecorp.armeria.common.HttpData;
+
+/**
+ * Assertion methods for HttpData.
+ */
+public final class HttpDataAssert extends AbstractResponseAssert<HttpData> {
+    HttpDataAssert(HttpData actual, TestHttpResponse response) {
+        super(actual, response);
+    }
+
+    /**
+     * Asserts that the actual {@link HttpData} is empty.
+     */
+    public TestHttpResponse isEmpty() {
+        assertTrue(actual().isEmpty());
+        return response();
+    }
+
+    /**
```

Comment:
I'm not sure how useful the byte array methods will be at the moment 😅 
What do you think of trying to apply the assertions in the armeria code base and adding it if needed? (in the interest of not adding too many public apis)

Reply:
Thank you for the suggestion! I will go ahead and remove the byte array-related methods.